test/metabase/pulse/render/png_test.clj

```diff
@@ -13,9 +20,64 @@
+    (let [^BufferedImage png (#'png/render-to-png test-table-html-1 1200)]
       ;; Check that width is within a range, since actual rendered result can very slightly by environment
       (is (< 140 (.getWidth png) 210))))
   (testing "The PNG of a table should not clip any of its content"
-    (let [^java.awt.image.BufferedImage png (@#'png/render-to-png test-table-html-2 1200)]
+    (let [^BufferedImage png (#'png/render-to-png test-table-html-2 1200)]
       (is (< 320 (.getWidth png) 360)))))
+
+(deftest installed-fonts-test
+  (testing "Are the correct fonts available for rendering?"
+    (is (contains?
+         (into #{} (map #(.getName ^Font %)) (.getAllFonts (GraphicsEnvironment/getLocalGraphicsEnvironment)))
+         "Lato Regular"))))
+
+(defn- bytes->image
+  [bytes]
+  (let [input-stream (ByteArrayInputStream. bytes)]
+    (ImageIO/read input-stream)))
+
+(defn- render-without-wrapping
+  [content width]
+  (-> [:html
+       [:body {:style (style/style
+                       {:font-family      "Lato, 'Helvetica Neue', 'Lucida Grande', sans-serif"
+                        :margin           0
+                        :padding          0
+                        :background-color :white})}
+                                 content]]
+      hiccup/html
+      (#'png/render-to-png width)))
+
+(defn- render-with-wrapping
+  [content width]
+  (-> {:content     content
+       :attachments {}}
+      (png/render-html-to-png width)
+      bytes->image))
+
+(deftest wrap-non-lato-characters-test
+  (testing "HTML Content inside tables with characters not supported by the Lato font are wrapped in a span."
+    (is (= [:td {:not-wrapped-in-here "안녕"}
+            [:span {:style "font-family: sans-serif;"} "안녕"]]
+           (#'png/wrap-non-lato-chars [:td {:not-here "안녕"} "안녕"])))
+    (is (= [:table
+            [:tr
+             [:td "this is all Lato-compatible, baby!"]
+             [:td [:span {:style "font-family: sans-serif;"} "This part's English. This part is 英語ではありません"]]]]
+           (#'png/wrap-non-lato-chars
+            [:table
+             [:tr
+              [:td "this is all Lato-compatible, baby!"]
+              [:td "This part's English. This part is 英語ではありません"]]])))))
+
+(deftest non-lato-characters-can-render-test
+  (testing "Strings containing characters that are not included in the Lato font can still be rendered."
+    (let [content                       [:span "안녕"]
+          ^BufferedImage broken-render  (render-without-wrapping content 200)
+          ^BufferedImage working-render (render-with-wrapping content 200)]
+      ;; The broken-render's width is around 17px. It is the width of 2 `[?]` charaters
+      ;; We assert that the working render is wider based on the assumption (verified manually by
+      ;; actually looking at the rendered images) that the correctly rendered glyphs are wider.
+      (is (< (.getWidth broken-render) (.getWidth working-render))))))
```

Comment:
For a visual ref, here's the broken/working render:

<img width="835" alt="image" src="https://github.com/metabase/metabase/assets/21064735/91690d3b-6b5e-4ca8-a778-3ec92def2160">